scripts/zkai_d128_proof_native_two_slice_compression_gate.py

```
@@ -0,0 +1,1065 @@
+
+
+def write_outputs(payload: dict[str, Any], json_path: pathlib.Path | None, tsv_path: pathlib.Path | None) -> None:
+    validate_payload(payload)
+    outputs: list[tuple[pathlib.Path, bytes]] = []
+    if json_path is not None:
+        outputs.append((_safe_output_path(json_path, ".json"), json.dumps(payload, indent=2, sort_keys=True).encode("utf-8") + b"\n"))
+    if tsv_path is not None:
+        outputs.append((_safe_output_path(tsv_path, ".tsv"), to_tsv(payload).encode("utf-8")))
+    resolved_outputs = [path.resolve(strict=False) for path, _data in outputs]
+    if len(resolved_outputs) != len(set(resolved_outputs)):
+        raise D128ProofNativeTwoSliceCompressionError("write-json and write-tsv output paths must be distinct")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)